executor/window.go

```diff
@@ -337,7 +337,15 @@ func (p *rowFrameWindowProcessor) appendResult2Chunk(ctx sessionctx.Context, row
 			if slidingWindowAggFunc != nil && initializedSlidingWindow {
 				err = slidingWindowAggFunc.Slide(ctx, rows, lastStart, lastEnd, shiftStart, shiftEnd, p.partialResults[i])
 			} else {
-				_, err = windowFunc.UpdatePartialResult(ctx, rows[start:end], p.partialResults[i])
+				// For MinMaxSlidingWindowAggFuncs, it needs the absolute value of each start of window, to compare
+				// whether elements inside deque are out of current window.
+				if minMaxSlidingWindowAggFunc, ok := windowFunc.(aggfuncs.MaxMinSlidingWindowAggFunc); ok {
```

Comment:
The `maxMin4XXXSliding` implements `SlidingWindowAggFunc` interface also,  the code can run to here?

Reply:
Yes, code could run to here. 

In this PR, `maxMin4XXXSliding` implements 3 interfaces:

![image](https://user-images.githubusercontent.com/34013484/106736856-f17f4a00-6650-11eb-88c8-fc89450d7497.png)

We still need `SlidingWindowAggFunc` for `Slide` method to run sliding window function. But in deque-based algo, I need the absolute index of window start, but directly changing `UpdatePartialResult()` signature is cubersome, so I came up with a compromise solution, setting window start index inside a `windowInfo` which is composited with all `maxMin4XXXSliding` struct. 

In order to distinguish `maxMin4XXXSliding` from other `aggFuncs`, I implemented `MaxMinSlidingWindowAggFunc` for them. Therefore during window initialization(window.go:L339-L449), all `maxMin4XXXSliding` could be picked up by asserting type of `MaxMinSlidingWindowAggFunc` and pass in window related info.

Reply:
Thanks for your explanation, I already understand your solution.